Improve composer version defaults

AaronFeledy opened the pull request:
Some newer things, e.g. Symfony 7 Console, aren't compatible with old composer. Old PHP is not compatible with new composer.

This PR updates the default Composer version from 2.2.22 to 2.8.2, with additional logic to set the defaults to the appropriate LTS versions for older PHP versions.

I also updated patch versions of Composer in the images.

reynoldsalec commented:
That sounds like a great idea. Scratching my head on why the tests are borking with the "UNeedSetup", was noticing on that on a little PR I was working on.

AaronFeledy commented:
I think it's something with the latest edge version. This should allow us to manually run tests against alternate release channels once it's merged: #138

reynoldsalec commented:
Yeah something is definitely busted on edge, getting the same error across a bunch of different plugins.

AaronFeledy commented:
This is now working as expected. Composer 2.8.2 has since been released so I updated that to be the default for newer PHP versions. I also found and fixed a bug where `compser_version: 1` with a number instead of a string would always install 2.2.x.

I left the preinstalled versions in the images alone for now and will address those in #119
